xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-rest/src/main/java/org/xwiki/annotation/rest/internal/AnnotationsRESTResource.java

```diff
@@ -162,20 +172,30 @@ protected AnnotationResponse postAnnotation(DocumentReference documentReference,
             // Initialize the context with the correct value.
             updateContext(documentReference);
 
-            String documentName = referenceSerializer.serialize(documentReference);
+            String documentName = this.referenceSerializer.serialize(documentReference);
 
             // check access to this function
-            if (!annotationRightService.canAddAnnotation(documentName, getXWikiUser())) {
+            if (!this.annotationRightService.canAddAnnotation(documentName, getXWikiUser())) {
                 throw new WebApplicationException(Status.UNAUTHORIZED);
             }
 
             // add the annotation
-            Map<String, Object> annotationMetadata = getMap(request.getAnnotation());
-            annotationService.addAnnotation(documentName, request.getSelection(), request.getSelectionContext(),
+            Map<String, Object> annotationsMap = getMap(request.getAnnotation());
+
+            Map<String, Object> annotationMetadata = new HashMap<>();
+            annotationMetadata.put(DATE_KEY, annotationsMap.get(DATE_KEY));
+            annotationMetadata.put(AUTHOR_KEY, annotationsMap.get(AUTHOR_KEY));
+            if (annotationsMap.containsKey("RequiresHTMLConversion")) {
```

Comment:
what is this key?

Reply:
Note that I don't know how the annotation code works. Just curious to understand the logic. What is annotationMetadata?